cg/services/run_devices/abstract_classes.py

```diff
@@ -83,6 +87,27 @@ def post_process(self, run_name: str, dry_run: bool = False):
         """Store sequencing metrics in StatusDB and relevant files in Housekeeper."""
         pass
 
+    def post_process_all(self, run_names: list[str], dry_run: bool = False) -> bool:
+        """Post-process all PacBio runs in the sequencing directory."""
+        LOG.info("Starting post-processing for all runs")
+        is_post_processing_successful = True
+        for run_name in run_names:
+            if self._is_run_post_processed(run_name):
+                LOG.info(f"Run {run_name} has already been post-processed")
+                continue
+            try:
+                self.post_process(run_name=run_name, dry_run=dry_run)
+            except Exception as error:
+                LOG.error(f"Could not post-process {self.instrument} run {run_name}: {error}")
+                is_post_processing_successful = False
+        if not is_post_processing_successful:
+            raise PostProcessingError(f"Could not post-process all {self.instrument} runs")
+
+    def _is_run_post_processed(self, run_name: str) -> bool:
+        """Check if a run has been post-processed."""
+        processing_complete_file = Path(self.sequencing_dir, run_name, POST_PROCESSING_COMPLETED)
+        return processing_complete_file.exists()
+
```

Comment:
Also both of these functions access a `self.sequencing_dir` and `self.insturment` This is a concrete implementation that requires all the interfaces that implement this ABC to have these parameters. we should not do this, as it introduces a dependecy.

Reply:
I think it is the base of post-processing services to depend on a `sequencing_dir` where to find the runs to post-process. 

Regarding the `self.instrument`, I don't see a problem with having it but it is not crucial, it is just for the logs, it could be removed